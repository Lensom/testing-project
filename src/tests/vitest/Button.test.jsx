import { render, screen, fireEvent } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import CustomButton from '@components/Button';

describe('Testing Button component', () => {
  it('Click on button', async () => {
    const { container } = render(<CustomButton />);

    const button = screen.getByText('Plus');
    const counter = container.querySelector('p');

    expect(counter.textContent).toBe('Counter: 0');

    fireEvent.click(button);

    expect(counter.textContent).toBe('Counter: 1');
  });

  it('Matches snapshot', () => {
    const { container } = render(<CustomButton />);

    expect(container).toMatchSnapshot();
  });

  it('Sending prop day to compononent', async () => {
    const { container } = render(<CustomButton day={'01.01.2025'} />);
    const currentDay = container.querySelector('span');

    expect(currentDay.textContent).toBe('Current day: 01.01.2025');
  });
});
