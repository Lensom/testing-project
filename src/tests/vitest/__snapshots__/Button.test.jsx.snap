// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing Button component > Matches snapshot 1`] = `
<div>
  <button>
    Plus
  </button>
  <p>
    Counter: 
    0
  </p>
  <span>
    Current day: 
  </span>
</div>
`;
